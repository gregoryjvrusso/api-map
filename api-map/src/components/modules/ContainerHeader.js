import React, { useState, useEffect } from 'react'
import axios from 'axios'
import styled from 'styled-components'
import Button from '../elements/Button'
import InputCEP from '../elements/InputCEP'
import Label from '../elements/Label'

const ContainerHeaderElements = styled.div`
  background-color: ${p => p.theme.colors.backgrounds.primary.gray};
  display: flex;
  flex-direction: column;
  padding: 0.5em;
`
const TitleHeader = styled.p`
  font-weight: bold;
`
const MessageError = styled.p`
  color: red;
  margin: 0;
`
const ContainerHeader = ({ setData }) => {
  const [cep, setCep] = useState('')
  const [message, setMessage] = useState(false)
  const fetchData = async (url) => {
    try {
      const result = await axios(url)
      if (result.data.erro) {
        setData('')
        setMessage(true)
      } else {
        setData(result.data)
        setMessage(false)
      }
    } catch (error) {
      console.log(error)
    }
  }

  const validatorCep = (cep) => {
    const regex = RegExp(/^\d{5}-\d{3}$/)
    if (regex.test(cep)) {
      let cepCheck = cep.replace('-', '')
      fetchData(`https://viacep.com.br/ws/${cepCheck}/json/`)
    } else {
      setMessage(true)
    }
  }
  return (
    <ContainerHeaderElements>
      <TitleHeader>Consultar</TitleHeader>
      {message && <MessageError>Cep inv√°lido</MessageError>}
      <form onSubmit={e => {
        e.preventDefault()
        validatorCep(cep)
      }}>
        <Label htmlFor='cep'>
          CEP
        </Label>
        <InputCEP
          width={['70%', '15em', '15em']}
          name='cep'
          value={cep}
          onChange={(e) => {
            setMessage(false)
            setCep(e.target.value)
          }}
          id='cep'
          type='text'
          placeholder='02250-250'
        />
        <Button width={['95%', '8em', '8em']} type='submit' primary>
          Buscar
        </Button>
      </form>
    </ContainerHeaderElements>
  )
}

export default ContainerHeader