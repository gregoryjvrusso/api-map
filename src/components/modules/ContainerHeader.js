import React, { useState } from 'react'
import styled from 'styled-components'
import { theme } from 'styled-tools'
import Form from '../elements/Form/Form'
import Text from '../elements/Text/Text'

const ContainerHeaderElements = styled.div`
  background-color: ${theme('colors.backgrounds.primary.gray')};
  display: flex;
  flex-direction: column;
  padding: 0.5em;
`

const ContainerHeader = ({ setData }) => {
  const [cep, setCep] = useState('')
  const [message, setMessage] = useState(false)

  return (
    <ContainerHeaderElements>
      <Text
        text={'Consultar'}
        tag={'h1'}
        fontSize='1em'
      />
      {message && <Text text={'Cep invÃ¡lido'} warning />}
      <Form
        setCep={setCep}
        setMessage={setMessage}
        setData={setData}
        cep={cep}
      />
    </ContainerHeaderElements>
  )
}

export default ContainerHeader