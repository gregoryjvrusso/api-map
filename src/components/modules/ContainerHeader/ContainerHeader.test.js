import React from 'react'
import renderer from 'react-test-renderer'
import ContainerHeader from './ContainerHeader.js'

const mock = {
  "cep": "01001-000",
  "logradouro": "Praça da Sé",
  "complemento": "lado ímpar",
  "bairro": "Sé",
  "localidade": "São Paulo",
  "uf": "SP",
  "unidade": "",
  "ibge": "3550308",
  "gia": "1004"
}

describe('Container Header component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const render = renderer
      .create(<ContainerHeader data={mock} />)
      .toJSON()
    expect(render).toMatchSnapshot()
  })
})
