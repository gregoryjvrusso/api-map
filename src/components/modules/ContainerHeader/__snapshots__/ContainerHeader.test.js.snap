// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Header component deve conseguir renderizar [snapshot] 1`] = `
<div
  className="sc-ifAKCX fNSzqa"
>
  <h1
    className="sc-bxivhb dGpotq"
    fontSize="1em"
  >
    Consultar
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      className="sc-bdVaJa FTQhB"
      htmlFor="cep"
    >
      CEP
    </label>
    <input
      className="sc-bwzfXH fRVSPy"
      data-testid="form-input"
      id="cep"
      name="cep"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onPaste={[Function]}
      placeholder="02250-250"
      type="text"
      value=""
      width={
        Array [
          "70%",
          "15em",
          "15em",
        ]
      }
    />
    <button
      className="sc-htpNat irRmst"
      data-testid="form-buttons"
      type="submit"
      width={
        Array [
          "95%",
          "8em",
          "8em",
        ]
      }
    >
      Buscar
    </button>
  </form>
</div>
`;
